flake.nix

```
@@ -127,10 +127,13 @@
           checks = self'.checks;
           inputsFrom = [ typst ];
 
-          buildInputs = with pkgs; [
+          buildInputs = with rust-toolchain; [
             rust-analyzer
+            rust-src
           ];
```

Since `with pkgs` is the usual here, let's prefer to be explicit.

```suggestion
          buildInputs = [
            rust-toolchain.rust-analyzer
            rust-toolchain.rust-src
          ];
```